packages/devtools_server/lib/src/external_handlers.dart

```diff
@@ -97,7 +100,18 @@ Future<shelf.Handler> defaultHandler(
     if (debugMode) {
       return debugProxyHandler!(request);
     } else {
-      return buildDirHandler!(request);
+      // If the user hits Refresh when on a page being handled by the Flutter
```

Comment:
The changes to this file will need to be applied to the g3 handler as well, but I have verified that things work as expected when that is done. I'll update the g3 handler when this change rolls into g3.